### ISSUE TITLE
Option to disable spinner

### ISSUE THREAD

**Issue by Jookia** [boxed/mutmut#250]
The spinner causes a huge log output in CI like:

```

⠋ Running...
⠙ Running...
⠹ Running...
⠸ Running...
⠼ Running...
⠴ Running...
⠦ Running...
⠧ Running...
⠇ Running...
⠏ Running...
⠋ Running...
⠙ Running...
⠹ Running...
⠸ Running...
⠼ Running...
⠴ Running...
⠦ Running...
⠧ Running...
⠇ Running...
⠏ Running...
⠋ Running...
⠙ Running...
⠹ Running...
⠸ Running...
⠼ Running...
⠴ Running...
⠦ Running...
⠧ Running...
⠇ Running...
⠏ Running...
⠋ Running...
⠙ Running...
⠹ Running...
⠸ Running...
⠼ Running...
⠴ Running...
⠦ Running...
⠧ Running...
⠇ Running...
⠏ Running...
⠋ Running...
⠙ Running...
⠹ Running...
⠸ Running...
⠼ Running...
⠴ Running...
⠦ Running...
⠧ Running...
⠇ Running...
⠏ Running...
⠋ Running...
⠙ Running...
⠹ Running...
⠸ Running...
```

Additionally it's unclear what it means since it isn't valid Braille.

An option to disable this would be good.

**Comment by boxed** [comment]
`--no-progress`

Isn't that shown when you do --help?

**Comment by Jookia** [comment]
--help gives this:

```
Usage: mutmut [OPTIONS] COMMAND [ARGS]...

  Mutation testing system for Python.

Options:
  -h, --help  Show this message and exit.

Commands:
  apply       Apply a mutation on disk.
  html        Generate a HTML report of surviving mutants.
  junitxml    Show a mutation diff with junitxml format.
  result-ids  Print the IDs of the specified mutant classes (separated by...
  results     Print the results.
  run         Runs mutmut.
  show        Show a mutation diff.
  version     Show the version and exit.
```

Is there more help somewhere?

**Comment by alexandrebarbaruiva** [comment]
@Jookia use `mutmut run --no-progress`

**Comment by boxed** [comment]
@alexandrebarbaruiva he says that didn't work

**Comment by DudeNr33** [comment]
Help for the individual commands is available through `mutmut <command> --help`:

```
D:\>mutmut run --help
Usage: mutmut run [OPTIONS] [ARGUMENT]

  Runs mutmut. You probably want to start with just trying this. If you supply
  a mutation ID mutmut will check just this mutant.

Options:
  --paths-to-mutate TEXT
  --disable-mutation-types TEXT   Skip the given types of mutations.
  --enable-mutation-types TEXT    Only perform given types of mutations.
  --paths-to-exclude TEXT
  --runner TEXT
  --use-coverage
  --use-patch-file TEXT           Only mutate lines added/changed in the given
                                  patch file
  --rerun-all                     If you modified the test_command in the
                                  pre_mutation hook, the default test_command
                                  (specified by the "runner" option) will be
                                  executed if the mutant survives with your
                                  modified test_command.
  --tests-dir TEXT
  -m, --test-time-multiplier FLOAT
  -b, --test-time-base FLOAT
  -s, --swallow-output            turn off output capture
  --dict-synonyms TEXT
  --pre-mutation TEXT
  --post-mutation TEXT
  --simple-output                 Swap emojis in mutmut output to plain text
                                  alternatives.
  --no-progress                   Disable real-time progress indicator
  -h, --help                      Show this message and exit.
```

> Additionally it's unclear what it means since it isn't valid Braille.

😄 👍

**Comment by alexandrebarbaruiva** [comment]
@boxed it seems @Jookia was using `mutmut --no-progress` instead of  `mutmut run --no-progress` due to him posting `--help` from `mutmut --help` instead of `mutmut run --help`

**Comment by Jookia** [comment]
mutmut run --no-progress does not remove the spinner.

**Comment by Jookia** [comment]
I spent a while trying to 'clean up' mutmut's output to be usable in CI, but because mutmut always assumes the output is a terminal it will cause havoc with programs that buffer lines of text at a time, such as most Unix programs. It's extremely disappointing to see the lack of basic line-based output in a terminal application, not just for CI but for the amount of wear it causes on output displays for no reason.

**Comment by boxed** [comment]
What? Haha. Wear on output displays?! Is this 1972?

**Comment by Jookia** [comment]
> What? Haha. Wear on output displays?! Is this 1972?

Uh, Braille displays? The very expensive devices with moving components that you're ensuring update every second or so?

**Comment by boxed** [comment]
Aha. Well you should have been more specific. Almost all output displays are screens. 

If the spinner still goes with the no output flag that's a bug and should be fixed. But let's not get out if hand here. This is a hobby project of mine that i just don't have time to work on with two small kids. 

I welcome PRs.

**Comment by Jookia** [comment]
What are the chances of a PR being merged? There's a lot open at the
moment.

**Comment by boxed** [comment]
For something simple? Extremely high.

**Comment by greenlynx** [comment]
I don't have time to raise a PR at the moment but it looks like line 975 in `mutmut/__init.py__` is the culprit here. The `register` method on the `Progress` class prints progress without checking the `--no-progress` config value, so even though fewer lines of output get written with this option enabled, a new line is still written each time the progress is updated:

```python
    def register(self, status):
        if status == BAD_SURVIVED:
            self.surviving_mutants += 1
        elif status == BAD_TIMEOUT:
            self.surviving_mutants_timeout += 1
        elif status == OK_KILLED:
            self.killed_mutants += 1
        elif status == OK_SUSPICIOUS:
            self.suspicious_mutants += 1
        elif status == SKIPPED:
            self.skipped += 1
        else:
            raise ValueError('Unknown status returned from run_mutation: {}'.format(status))
        self.progress += 1
        self.print() # <========= this prints the status regardless of the config setting
```

**Comment by boxed** [comment]
Released 2.4.1 with this change.

**Comment by cwegener** [comment]
Has this option been removed in 3.x? I always see the spinner / progress again regardless of a tty being attached.
